File: app/src/main/java/com/hypnoweb/hypnowebapp/di/modules/ActivityModule.java
package com.hypnoweb.hypnowebapp.di.modules;


import com.hypnoweb.hypnowebapp.splashscreen.SplashScreenActivity;

import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityModule {

    @ContributesAndroidInjector(modules = FragmentBuilderModule.class)
    @ActivityKey(SplashScreenActivity.class)
    abstract SplashScreenActivity contributeSplashActivity();
}

File: app/src/main/java/com/hypnoweb/hypnowebapp/di/modules/ActivityModule.kt
package com.hypnoweb.hypnowebapp.di.modules

import com.hypnoweb.hypnowebapp.ui.catalog.CatalogActivity
import com.hypnoweb.hypnowebapp.ui.splashscreen.SplashScreenActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector


@Module
abstract class ActivityModule {

    @ContributesAndroidInjector(modules = arrayOf(FragmentBuilderModule::class))
    abstract fun provideSplashScreenActivity(): SplashScreenActivity

    @ContributesAndroidInjector(modules = arrayOf())
    abstract fun provideCatalogActivity(): CatalogActivity

}